vault/values.yaml

```diff
@@ -190,9 +190,9 @@ vault:
 
     # Check https://www.vaultproject.io/docs/configuration/storage/ for supported storage backends.
     # @ignored
-    storage:
-      file:
-        path: "/vault/file"
+    storage: {}
+      # file:
+      #   path: "/vault/file"
```

Comment:
Also revert this please

Reply:
I did some testing with your setup, and I verified that, the missing conditional was the error here that was setting the false file storage.

## I have done the following manual tests

NOTE: after each test I was running the following command:

```bash
helm template vault ./ -n vault -f ./values.yaml >  vault.yaml
```

(showing only the vault-config secret)

### 1. Not changing anything

```yaml
storage:
  file:
    path: "/vault/file"
```

```yaml
---
# Source: vault/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: "vault-config"
  namespace: vault
  labels:
    helm.sh/chart: vault-0.0.0
    app.kubernetes.io/name: "vault"
    app.kubernetes.io/instance: "vault"
    app.kubernetes.io/managed-by: "Helm"
data:
  config.json:
    ewogICJkaXNhYmxlX21sb2NrIjogZmFsc2UsCiAgImxpc3RlbmVyIjogewogICAgInRjcCI6IHsKICAgICAgImFkZHJlc3MiOiAiWzo6XTo4MjAwIiwKICAgICAgInRsc19jZXJ0X2ZpbGUiOiAiL3ZhdWx0L3Rscy9zZXJ2ZXIuY3J0IiwKICAgICAgInRsc19rZXlfZmlsZSI6ICIvdmF1bHQvdGxzL3NlcnZlci5rZXkiCiAgICB9CiAgfSwKICAic3RvcmFnZSI6IHsKICAgICJmaWxlIjogewogICAgICAicGF0aCI6ICIvdmF1bHQvZmlsZSIKICAgIH0KICB9LAogICJ0ZWxlbWV0cnkiOiB7CiAgICAic3RhdHNkX2FkZHJlc3MiOiAibG9jYWxob3N0OjkxMjUiCiAgfSwKICAidWkiOiB0cnVlCn0=
  vault-config.yml:
    YXV0aDoKLSByb2xlczoKICAtIGJvdW5kX3NlcnZpY2VfYWNjb3VudF9uYW1lczogZGVmYXVsdAogICAgYm91bmRfc2VydmljZV9hY2NvdW50X25hbWVzcGFjZXM6IGRlZmF1bHQKICAgIG5hbWU6IGRlZmF1bHQKICAgIHBvbGljaWVzOiBhbGxvd19zZWNyZXRzCiAgICB0dGw6IDFoCiAgdHlwZToga3ViZXJuZXRlcwpwb2xpY2llczoKLSBuYW1lOiBhbGxvd19zZWNyZXRzCiAgcnVsZXM6IHwKICAgIHBhdGggInNlY3JldC8qIiB7CiAgICAgIGNhcGFiaWxpdGllcyA9IFsiY3JlYXRlIiwgInJlYWQiLCAidXBkYXRlIiwgImRlbGV0ZSIsICJsaXN0Il0KICAgIH0Kc2VjcmV0czoKLSBkZXNjcmlwdGlvbjogR2VuZXJhbCBzZWNyZXRzLgogIG9wdGlvbnM6CiAgICB2ZXJzaW9uOiAyCiAgcGF0aDogc2VjcmV0CiAgdHlwZToga3Y=
---
```

Decoded `config.json` field

```json
{
  "disable_mlock": false,
  "listener": {
    "tcp": {
      "address": "[::]:8200",
      "tls_cert_file": "/vault/tls/server.crt",
      "tls_key_file": "/vault/tls/server.key"
    }
  },
  "storage": {
    "file": {
      "path": "/vault/file"
    }
  },
  "telemetry": {
    "statsd_address": "localhost:9125"
  },
  "ui": true
}
```

### 2. Changing storage to null

```yaml
storage: null
```

```yaml
---
# Source: vault/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: "vault-config"
  namespace: vault
  labels:
    helm.sh/chart: vault-0.0.0
    app.kubernetes.io/name: "vault"
    app.kubernetes.io/instance: "vault"
    app.kubernetes.io/managed-by: "Helm"
data:
  config.json:
  vault-config.yml:
    YXV0aDoKLSByb2xlczoKICAtIGJvdW5kX3NlcnZpY2VfYWNjb3VudF9uYW1lczogZGVmYXVsdAogICAgYm91bmRfc2VydmljZV9hY2NvdW50X25hbWVzcGFjZXM6IGRlZmF1bHQKICAgIG5hbWU6IGRlZmF1bHQKICAgIHBvbGljaWVzOiBhbGxvd19zZWNyZXRzCiAgICB0dGw6IDFoCiAgdHlwZToga3ViZXJuZXRlcwpwb2xpY2llczoKLSBuYW1lOiBhbGxvd19zZWNyZXRzCiAgcnVsZXM6IHwKICAgIHBhdGggInNlY3JldC8qIiB7CiAgICAgIGNhcGFiaWxpdGllcyA9IFsiY3JlYXRlIiwgInJlYWQiLCAidXBkYXRlIiwgImRlbGV0ZSIsICJsaXN0Il0KICAgIH0Kc2VjcmV0czoKLSBkZXNjcmlwdGlvbjogR2VuZXJhbCBzZWNyZXRzLgogIG9wdGlvbnM6CiAgICB2ZXJzaW9uOiAyCiAgcGF0aDogc2VjcmV0CiAgdHlwZToga3Y=
---
```

3. Changing storage to your setup:

```yaml
storage:
  dynamodb:
    ha_enabled: true
    region: us-east-1
    table: "vault-backend"
```

```yaml
---
# Source: vault/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: "vault-config"
  namespace: vault
  labels:
    helm.sh/chart: vault-0.0.0
    app.kubernetes.io/name: "vault"
    app.kubernetes.io/instance: "vault"
    app.kubernetes.io/managed-by: "Helm"
data:
  config.json:
    ewogICJkaXNhYmxlX21sb2NrIjogZmFsc2UsCiAgImxpc3RlbmVyIjogewogICAgInRjcCI6IHsKICAgICAgImFkZHJlc3MiOiAiWzo6XTo4MjAwIiwKICAgICAgInRsc19jZXJ0X2ZpbGUiOiAiL3ZhdWx0L3Rscy9zZXJ2ZXIuY3J0IiwKICAgICAgInRsc19rZXlfZmlsZSI6ICIvdmF1bHQvdGxzL3NlcnZlci5rZXkiCiAgICB9CiAgfSwKICAic3RvcmFnZSI6IHsKICAgICJkeW5hbW9kYiI6IHsKICAgICAgImhhX2VuYWJsZWQiOiB0cnVlLAogICAgICAicmVnaW9uIjogInVzLWVhc3QtMSIsCiAgICAgICJ0YWJsZSI6ICJ2YXVsdC1iYWNrZW5kIgogICAgfQogIH0sCiAgInRlbGVtZXRyeSI6IHsKICAgICJzdGF0c2RfYWRkcmVzcyI6ICJsb2NhbGhvc3Q6OTEyNSIKICB9LAogICJ1aSI6IHRydWUKfQ==
  vault-config.yml:
    YXV0aDoKLSByb2xlczoKICAtIGJvdW5kX3NlcnZpY2VfYWNjb3VudF9uYW1lczogZGVmYXVsdAogICAgYm91bmRfc2VydmljZV9hY2NvdW50X25hbWVzcGFjZXM6IGRlZmF1bHQKICAgIG5hbWU6IGRlZmF1bHQKICAgIHBvbGljaWVzOiBhbGxvd19zZWNyZXRzCiAgICB0dGw6IDFoCiAgdHlwZToga3ViZXJuZXRlcwpwb2xpY2llczoKLSBuYW1lOiBhbGxvd19zZWNyZXRzCiAgcnVsZXM6IHwKICAgIHBhdGggInNlY3JldC8qIiB7CiAgICAgIGNhcGFiaWxpdGllcyA9IFsiY3JlYXRlIiwgInJlYWQiLCAidXBkYXRlIiwgImRlbGV0ZSIsICJsaXN0Il0KICAgIH0Kc2VjcmV0czoKLSBkZXNjcmlwdGlvbjogR2VuZXJhbCBzZWNyZXRzLgogIG9wdGlvbnM6CiAgICB2ZXJzaW9uOiAyCiAgcGF0aDogc2VjcmV0CiAgdHlwZToga3Y=
---
```

Decoded `config.json` field:

```json
{
  "disable_mlock": false,
  "listener": {
    "tcp": {
      "address": "[::]:8200",
      "tls_cert_file": "/vault/tls/server.crt",
      "tls_key_file": "/vault/tls/server.key"
    }
  },
  "storage": {
    "dynamodb": {
      "ha_enabled": true,
      "region": "us-east-1",
      "table": "vault-backend"
    }
  },
  "telemetry": {
    "statsd_address": "localhost:9125"
  },
  "ui": true
}
```

Reply:
By my set up you mean you tested using the vault chart on the test repo I created on https://github.com/MuriukiPM/helm-charts-playground/blob/vault/README.md where this chart is a dependency?
I am getting different results to yours using that test repo to validate reverting this change. That is if I build this chart locally with helm package then locally still, replace the vault/charts/vault-1.20.4.tgz in the test repo with the built chart and run helm template

Reply:
No, by your setup I meant the changes I made to the original code, including adding that conditional and modifying the storage you use, which I then tested with helm template.

Unsetting the storage as you suggested isn't viable, as the Helm Chart wouldn't function out of the box without specified storage.
If you have ideas for adding logic to the `secret.yaml` template that would define file storage as a default when nothing else is specified, I'd be glad to review those suggestions.
To address your issue, I recommend directly modifying the original values of the Helm Chart and building it yourself with the `make helm-chart` target.

Reply:
We can add back the changes from https://github.com/banzaicloud/banzai-charts/pull/215 since those address this issue of a default storage.

Reply:
AFAIK, this workaround was placed in because Helm was not dropping a value if it was set to null, but now it does, so I'm not convinced that this would be an ideal solution.

With Helm-Charts you usually set defaults the way we do right now, not by specifying it like it was with that workaround.